### PR TITLE
163148396 Clean up wrong change-dates from db

### PR DESCRIPTION
**Pull request by rckopteri**
# Added
* Add tests for 25 weeks and 35 weeks to ensure that 30 week detection window works
* I had to refactor detection test codes. I didn't change them. I only added two more.

### CODE REVIEW

File: database/src/main/resources/db/migration/V1_142__remove_old_change_dates.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- Remove gtfs-transit-changes.change-date dates from database to prevent unwanted change-detections
```
Comment (philosotractor) [inline database/src/main/resources/db/migration/V1_142__remove_old_change_dates.sql]: Doesn't this affect production as well? Current week's change dates would be lost, and re-created...when?

File: ote/test/clj/ote/transit_changes/detection_test_utilities.clj [hunk 2]
```diff
@@ -0,0 +1,27 @@
+(ns ote.transit-changes.detection-test-utilities
```
Comment (philosotractor) [inline ote/test/clj/ote/transit_changes/detection_test_utilities.clj]: Could code formatting be fixed, some of fields are far away on the right in the file.

File: ote/test/clj/ote/transit_changes/detection_window_test.clj [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+(deftest test-change-at-week-33
+	(let [result (-> data-change-at-week-33
+									 detection/changes-by-week->changes-by-route
+									 detection/detect-changes-for-all-routes)]
+		(is (= {:route-key test-utilities/route-name
+						:starting-week {:beginning-of-week (test-utilities/to-local-date 2019 1 14)
+														:end-of-week (test-utilities/to-local-date 2019 1 20)}} ;; HOX! No different week, no change
+					 (-> result
+							 first
+							 (select-keys [:no-traffic-start-date :no-traffic-end-date :route-key :no-traffic-change
```
Comment (philosotractor) [inline ote/test/clj/ote/transit_changes/detection_window_test.clj]: select-keys-detect-changes-for-all-routes

File: ote/test/clj/ote/transit_changes/detection_window_test.clj [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+									 detection/changes-by-week->changes-by-route
+									 detection/detect-changes-for-all-routes)]
+		(is (= {:route-key test-utilities/route-name
+						:starting-week {:beginning-of-week (test-utilities/to-local-date 2019 1 14)
+														:end-of-week (test-utilities/to-local-date 2019 1 20)}
+						:different-week {:beginning-of-week (test-utilities/to-local-date 2019 6 17)
+														 :end-of-week (test-utilities/to-local-date 2019 6 23)}}
+					 (-> result
+							 first
+							 (select-keys [:no-traffic-start-date :no-traffic-end-date :route-key :no-traffic-change
```
Comment (philosotractor) [inline ote/test/clj/ote/transit_changes/detection_window_test.clj]: select-keys-detect-changes-for-all-routes

File: database/src/main/resources/db/migration/V1_142__remove_old_change_dates.sql (hunk 1 above)
Comment (rckopteri) [reply database/src/main/resources/db/migration/V1_142__remove_old_change_dates.sql]: I assume that we will get this pr to production in the next week or two. Next correct detection cycle starts about 4 weeks from now. So if there is change-days that are in the next 7 days, they are generated with wrong code.

File: ote/test/clj/ote/transit_changes/detection_window_test.clj [hunk 5]
```diff
@@ -0,0 +1,55 @@
+(ns ote.transit-changes.detection-window-test
```
Comment (philosotractor) [inline ote/test/clj/ote/transit_changes/detection_window_test.clj]: "design for testability" i.e. should the tested functions take the detection window value as an argument.
That way tests could test using different values...in the other hand would that bring any value?

File: ote/test/clj/ote/transit_changes/detection_window_test.clj (hunk 3 above)
Comment (rckopteri) [reply ote/test/clj/ote/transit_changes/detection_window_test.clj]: fixed

File: ote/test/clj/ote/transit_changes/detection_window_test.clj (hunk 4 above)
Comment (rckopteri) [reply ote/test/clj/ote/transit_changes/detection_window_test.clj]: fixed

**Pull request merged**